### PR TITLE
Vampires can listen to the dead

### PR DESCRIPTION
**Pull request by adacovsk**
>Are we ever getting a VTM-themed vampire rework?
>Malkavian vampires sometimes hearing deadchat shitpost.
>`>midrounds when?`
>Truly the thoughts of the insane.

## What this does
- After 400 blood, you go crazy and hear the dead at random intervals

## Why it's good
A feature: it's new and exciting!

## Changelog

:cl:
 * rscadd: Vampires can occasionally listen to the dead

### CODE REVIEW

**Comment by Eneocho** [comment]
Unholy based.
An idea I've had for vampire specialization would be to pull some stuff straight outta VTM.
3 available clans stationside (since balancing and making the like 8 clans would be a nightmare).
Nosferatu: Ugly and blatant vampire, can't pretend to be human, access to stealth/covert mechanics such as invisibility and fake voices. Can screech, can mist form, can disease touch.
Malkavian: Insane. Can hear deadchat every now and then, which might be useful for insight on the round or just useless. Access to mind mechanics, like the cult monster visions talisman (forgot the name) and remote talk. Can enthrall, can hypnotize.
Gangrel: Animalistic. Can turn into spooky melee vampire monster and is very fast, can't use ranged weapons when monster mode but is very tanky to them as well. Can screech, can summon bats.
The 3 clans keep the nonspecific vampire bonuses such as healing, vampiric vision, cloak of darkness, etc.

You could pretend to be human with malkavian and gangrel, the former letting you fuck with people indirectly, the latter going full loud, while nosferatu letting you be spooky shadow vampire that everyone knows about but nobody knows where it is, for that paranoia feeling.

We already have most of the features already coded, such as the cult hallucinations for monster sight, transformation, invisibility, etc. We'd just need to implement it.

**Comment by Servalfang** [comment]
incredible pr

**Comment by jknpj** [comment]
>can read dead chat in regular intervals and without any gibberish to mess your understanding

That's not how malkavians or madness work, at all.

**Comment by ghost** [comment]
i have opened the mayor,
when the power drops, go in, and go in, and go in, and GO IN like the us marshal and his three daughters and we lose the vaccine

**Comment by adacovsk** [comment]
> > can read dead chat in regular intervals and without any gibberish to mess your understanding
> 
> That's not how malkavians or madness work, at all.

I never played the game, I'll add some rands

**Comment by Cartlord** [comment]
What if you heard the stuff being said by deadchat in the voices of other crew?

**Comment by luigil101** [comment]
> 

but then we'd all miss out on vampires hearing [ADMIN] dsay IC
Immulsions were meant to be shattered

**Comment by Eneocho** [comment]
> > > can read dead chat in regular intervals and without any gibberish to mess your understanding
> > 
> > 
> > That's not how malkavians or madness work, at all.
> 
> I never played the game, I'll add some rands

Malkavians can hear the madness network and in some cases the wraiths in an area. Wraiths are essentially the in-universe ghosts, while the madness network is the collective conscious/unconscious of the entire malkavian population. Due to the malkavians being insane (it's their vampiric defect), most of the network's "output" is either really convoluted, said in a really roundabout way or flat out gibberish ramblings a la Francis E Dec. Malkavians are, to some extent, precognitive, due to being instinctively able to access what other malkavians know, and what the network itself knows, which is canonically the consciousness of Malkav, the first malkavian and effectively a demigod.

You should play VTMB if tabletop rpgs aren't your cup of tea.

**Comment by Eneocho** [comment]
>I'll add some rands

Could we get an example of how it would look to a vampire?

**Comment by adacovsk** [comment]
> > I'll add some rands
> 
> Could we get an example of how it would look to a vampire?

Bro, I just stopped testing... It'll look like deadchat with the purple text, with salty ghosts like so:
Salty Clown moans, "TITS ASS FUCK"

I've updated the dead chat for vampires specifically so that it doesn't say:
(Follow) DEAD: Salty Clown moans, "TITS ASS FUCK"

**Comment by githubuser4141** [comment]
does what a vampire hear get logged? If not it'd be pretty easy for metabuddies to exploit? There should probably be some randomization/word limit to what's heard too so people can't spam paragraphs of detailed antag dox?

**Comment by adacovsk** [comment]
> does what a vampire hear get logged? If not it'd be pretty easy for metabuddies to exploit? There should probably be some randomization/word limit to what's heard too so people can't spam paragraphs of detailed antag dox?

Yes, the intervals and the amount of time hearing is randomized, so it's unlikely that it can be abused

**Comment by Kurfursten** [comment]
I have a feeling this will be regretted, but sometimes you have to let the emojis see why it's a bad idea firsthand

**Comment by Eneocho** [comment]
I can only see !!!FUN!!! as a result of this.

File: code/datums/gamemode/role/vampire_role.dm [hunk 1]
```diff
@@ -328,6 +331,21 @@
...
+		return
+	if(H.stat == DEAD)
+		return
+	if(locate(/datum/power/vampire/charisma) in current_powers && world.time > deadchat_timer)
+		deadchat = FALSE
+		H.client.prefs.toggles |= CHAT_DEAD
+		//have deadchat for 30 seconds every five minutes
+		spawn(rand(200, 400))
+			if(H.stat != DEAD)
+				deadchat_timer = world.time + 1800 + rand(300, 1200)
```
Comment (ShiftyRail) [inline code/datums/gamemode/role/vampire_role.dm:345]: there is this deadchat timer but you're not actually using again...

File: code/datums/gamemode/role/vampire_role.dm [hunk 2]
```diff
@@ -328,6 +331,21 @@
...
 		to_chat(C, "<span class='sinister'>Your heart is filled with dread, and you shake uncontrollably.</span>")
 
+/datum/role/vampire/proc/handle_deadspeak(var/mob/living/carbon/human/H)
+	if(!deadchat)
+		return
+	if(H.stat == DEAD)
+		return
+	if(locate(/datum/power/vampire/charisma) in current_powers && world.time > deadchat_timer)
+		deadchat = FALSE
+		H.client.prefs.toggles |= CHAT_DEAD
```
Comment (ShiftyRail) [inline code/datums/gamemode/role/vampire_role.dm:341]: bruv
don't touch client prefs outside of the client interface

File: code/modules/mob/say.dm [hunk 3]
```diff
@@ -75,21 +75,27 @@ var/list/global_deadchat_listeners = list()
...
+		if(M.client.holder && M.client.holder.rights & R_ADMIN) //admins can toggle deadchat on and off. This is a proc in admin.dm and is only give to Administrators and above
+			to_chat(M, rendered)//Admins can hear deadchat, if they choose to, no matter if they're blind/deaf or not.
+		else if(M.stat == DEAD && !istype(M, /mob/dead/observer/deafmute))
 			to_chat(M, rendered)
-		else if(M.client && istype(M,/mob/living/carbon/brain) && (M.client.prefs.toggles & CHAT_DEAD))
+		else if(istype(M,/mob/living/carbon/brain))
 			var/mob/living/carbon/brain/B = M
 			if(B.brain_dead_chat())
 				to_chat(M, rendered)
+		else if(isvampire(M))
```
Comment (ShiftyRail) [inline code/modules/mob/say.dm:96]: hahahahaha

**Comment by Eneocho** [comment]
I think these reviews came a bit late, shifty.

**Pull request merged**